components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1919,24 +1919,34 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         return subscription;
     }
 
-    protected async ensureIsUsageBasedFeatureFlagEnabled(user: User): Promise<void> {
-        const teams = await this.teamDB.findTeamsByUser(user.id);
-        const isUsageBasedBillingEnabled = await getExperimentsClientForBackend().getValueAsync(
-            "isUsageBasedBillingEnabled",
-            false,
-            {
-                user,
-                teams: teams,
-            },
+    protected async ensureStripeApisAllowed(sub: { user?: User; team?: Team }): Promise<void> {
+        this.ensureBillingMode(
+            sub,
+            // Stripe is allowed when you either are on the usage-based side already, or you can switch to)
+            (m) => m.mode === "usage-based" || (m.mode === "chargebee" && m.tier === "paid_cancelled_and_ubb"),
         );
-        if (!isUsageBasedBillingEnabled) {
-            throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "not allowed");
+    }
+
+    protected async ensureBillingMode(
+        sub: { user?: User; team?: Team },
+        pred: (m: BillingMode) => boolean,
```

Comment:
What does `pred` mean? 👀

Reply:
Predicate. Sorry, was a tad late when I wrote this :stuck_out_tongue: Will improve.